src/Components/ADT3DViewer/ADT3DViewer.tsx

```diff
@@ -471,6 +476,25 @@ const ADT3DViewerBase: React.FC<IADT3DViewerProps> = ({
         // // eslint-disable-next-line react-hooks/exhaustive-deps
     }, []);
 
+    const createBadge = (badgeGroup: SceneViewBadgeGroup) => {
```

Comment:
can we pull this out to a function and just pass in the callback for hover? we are close to being self contained.